### ISSUE TITLE
rework build system for version integrity 

### ISSUE THREAD

**Issue by muhamadazmy** [threefoldtech/zos#2026]
Right now when a release is created this happens:
- ZOS binaries are built and added to one flist (has version suffix)
- Runtime binaries are build (each in its own flist, has version suffix)

If the release name has (no-suffix, -qaX, or -rcX) suffix this happens:
- zos flist symlink is added to the `tf-zos` repo on the hub, with the proper network (rc maps to testing, and qa maps to quality). A no-suffix maps to production 
- symlinks from the runtime binary files are created under `tf-zos-v3-bins.<network>` (using same mapping of names as above) 

ZOS then track the version of the zos flist (the one that includes the zos binaries) under `tf-zos`. If a change was detected it will install that zos version and will make sure all runtime packages are the same as the one as the ones on the corresponding network repo (qa, testing, prodcution, or dev).

## The problem
the problem with this approach is that the version number has to have a suffix to force the build to put the files on the correct repo, which means we can't do a seamless version migration from say qa network to testing network without creating a new build (even if the same version). The build is not consistent since a release that fails to build zos binaries can still update it's runtime binaries hence the update will be half done without a way to rolling back.

## Solutions:
- Hub supporting (groups) of packages. This means that all binaries (runtime + zos) can be grouped under one `version` which means when the system update it needs to install all flists from that new group. This will require changes to the hub itself to support this. (recommended solution imho)
  - Basically a tag/group is created under tf-zos with the version name which below that has links to the actual build flists
- Hub supporting creating repos on the fly (that is not linked to specific account) but this is not clean and will pollute the hub repos page (don't really like this solution)
- Another solution would be to have a single flist that include EVERYTHING, this will complicate the build process so much and might not work because some binaries need to be built on specific platforms. (i would not go with this)

**Comment by muhamadazmy** [comment]
Work is already in progress on branch #2082 

So far the changes are to always build all binaries of a build under  `tf-autobuilder/{sha}` or `tf-autobuilder/{tag}` depends on if it's tagged or not.

Next when a version needed to be deployed a tag is created under `tf-zos` as follows `testing -> tf-autobuilds/{tag}` for example. This means to release the same version to qa, all what we need to do it to create the new tag `qa` to point to the same tf-autobuilds `tag`.

**Comment by muhamadazmy** [comment]
There is still some issues with the tagging so i contacted maxime to fix that in the github action itself.

**Comment by muhamadazmy** [comment]
Next step after that is
- [x] update `bootstrap` to use new style 
- [x] update `identityd` update procedure to use new style

**Comment by xmonader** [comment]
How far are we this one? @Eslam-Nawara @AbdelrahmanElawady it's your responsibility to have this properly documented :)
